router/transformer/transformer.go

```diff
@@ -209,62 +221,45 @@ func (trans *HandleT) ResponseTransform(ctx context.Context, responseData integr
+			//Detecting content type of the respBody
+			contentTypeHeader := strings.ToLower(http.DetectContentType(tempRespData))
+			//If content type is not of type "*text*", overriding it with empty string
+			if !(strings.Contains(contentTypeHeader, "text") ||
+				strings.Contains(contentTypeHeader, "application/json") ||
+				strings.Contains(contentTypeHeader, "application/xml")) {
+				tempRespData = []byte("")
 			}
-			// handling for 5xx
-			if resp.StatusCode >= 500 && resp.StatusCode < 600 {
-				var transError integrations.TransErrorT
-				rawTResponse := []byte(gjson.GetBytes(tempRespData, "output").Raw)
-				// if the response is not containing "output" is not an explicit error from transformer
-				if len(rawTResponse) != 0 {
-					err = json.Unmarshal(rawTResponse, &transError)
-					if err != nil {
-						respData = []byte("Failed to parse response transform at server error: " + err.Error())
-						respCode = 400
-						break
-					}
-					// error is explicitly lodged from transformer marking a destination side failure
-					if transError.ErrorDetailed.ResponseTransformFailure {
-						integrations.CollectDestErrorStats(rawTResponse)
-						respData = tempRespData
-						respCode = resp.StatusCode
-						break
-					}
-				}
-				// network level 5xx error we need to retry response transform request
-				trans.logger.Errorf("[Transformer Response Transform request failed] :: %+v", err)
-				requestFailed = true
-				if retryCount > maxRetry {
-					panic(fmt.Errorf("Transformer HTTP connection error: URL: %v Error: %+v", url, err))
-				}
-				retryCount++
-				time.Sleep(retrySleep)
-				//Refresh the connection
-				continue
+			if requestFailed {
+				trans.logger.Errorf("Failed request succeeded after %v retries, URL: %v", retryCount, url)
+			}
+			// response transform success
+			var transformerResponse integrations.TransResponseT
+			respData = []byte(gjson.GetBytes(tempRespData, "output").Raw)
+			integrations.CollectDestErrorStats(respData)
+			err = json.Unmarshal(respData, &transformerResponse)
+			// unmarshal failure
+			if err != nil {
+				errStr := string(respData) + " [Error at Response Transform, Unmarshaling::]" + err.Error()
+				trans.logger.Errorf(errStr)
+				respData = []byte(errStr)
+				respCode = http.StatusBadRequest
+				break
+			}
+			// unmarshal success
+			respData, err = json.Marshal(transformerResponse)
+			if err != nil {
+				panic(fmt.Errorf("[Response transform:: failed to Marshal transformer response : %+v", err))
 			}
-			rawTResponse := []byte(gjson.GetBytes(tempRespData, "output").Raw)
-			integrations.CollectDestErrorStats(rawTResponse)
-			respData = tempRespData
-			respCode = resp.StatusCode
+			respCode = int(transformerResponse.Status)
 			break
+
+		}
+
+		// fallback if both resp and err are nil
+		if resp == nil && err == nil {
+			panic(fmt.Errorf(`[Response Transform] Client returned nil response and nil error`))
 		}
```

Comment:
```suggestion
```
This condition could never happen,  either resp or err will be nil but never both.

> An error is returned if caused by client policy (such as CheckRedirect), or failure to speak HTTP (such as a network connectivity problem). A non-2xx status code doesn't cause an error.
> If the returned error is nil, the Response will contain a non-nil Body which the user is expected to close. If the Body is not both read to EOF and closed, the Client's underlying RoundTripper (typically Transport) may not be able to re-use a persistent TCP connection to the server for a subsequent "keep-alive" request.

Reply:
Got it,will address this